Assets/MRTK/Tests/PlayModeTests/InputRecordingSystemTests.cs

```diff
@@ -0,0 +1,177 @@
+        
+        [UnitySetUp]
+        public IEnumerator Init()
+        {
+            TestUtilities.InitializeMixedRealityToolkit(true);
+
+            yield return null;
+        }
+
+        [UnityTearDown]
+        public IEnumerator TearDown()
+        {
+            PlayModeTestUtilities.TearDown();
+
+            yield return null;
+        }
+
+        [UnityTest]
+        public IEnumerator TestHandRecording() => TestRecording(testHandsPath);
+        
+        [UnityTest]
+        public IEnumerator TestCameraRecording() => TestRecording(testCameraPath);
+        
+        [UnityTest]
+        public IEnumerator TestGazeRecording() => TestRecording(testGazePath);
+        
+        [UnityTest]
+        public IEnumerator TestAllRecording() => TestRecording(testAllPath);
+
+        private static IEnumerator TestRecording(string recordingProfilePath)
+        {
+            var profile = AssetDatabase.LoadAssetAtPath(profilePath, typeof(MixedRealityToolkitConfigurationProfile)) as MixedRealityToolkitConfigurationProfile;
+
+            MixedRealityToolkit.Instance.ActiveProfile = profile;
+
+            yield return null;
+            yield return null;
+
+            var inputSystem = PlayModeTestUtilities.GetInputSystem();
+            var recordingService = CoreServices.GetInputSystemDataProvider<InputRecordingService>();
+            var recordingProfile = AssetDatabase.LoadAssetAtPath(recordingProfilePath, typeof(MixedRealityInputRecordingProfile)) as MixedRealityInputRecordingProfile;
+
+            Debug.Log($"Record hand data: {recordingProfile.RecordHandData}");
+            Debug.Log($"Record camera data: {recordingProfile.RecordCameraPose}");
+            Debug.Log($"Record gaze data: {recordingProfile.RecordEyeGaze}");
+            
+            recordingService.InputRecordingProfile = recordingProfile;
+
+            if (recordingProfile.RecordEyeGaze)
+            {
+                CoreServices.InputSystem.EyeGazeProvider.Enabled = true;
+                CoreServices.InputSystem.EyeGazeProvider.UpdateEyeTrackingStatus(null, true);
+            }
+            
+            recordingService.StartRecording();
+
+            yield return MoveAround(recordingProfile.RecordHandData, recordingProfile.RecordCameraPose, recordingProfile.RecordEyeGaze);
+            
+            recordingService.StopRecording();
+            
+            string path = recordingService.SaveInputAnimation("TestRecording.bin", null);
+            var playbackService = CoreServices.GetInputSystemDataProvider<IMixedRealityInputPlaybackService>();
+
+            yield return null;
+
+            playbackService.LoadInputAnimation(path);
+
+            var animation = playbackService.Animation;
+            
+            Assert.True(recordingProfile.RecordHandData == animation.HasHandData && recordingProfile.RecordCameraPose == animation.HasCameraPose && recordingProfile.RecordEyeGaze == animation.HasEyeGaze);
+            playbackService.Play();
```

Comment:
Might be good to verify something such as that input actually did some movement. I could see the playback succeeding, but for example nothing actually moved (even though it should have).